### PR TITLE
[AIRFLOW-5632] Rename ComputeEngine hook

### PR DESCRIPTION
**Pull request by michalslowikowski00**
This is a part of AIP-21

---

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5632
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -163,7 +163,7 @@ The following table shows changes in import paths.
 |airflow.contrib.hooks.datastore_hook.DatastoreHook                                                                |airflow.gcp.hooks.datastore.DatastoreHook                                                                  |
 |airflow.contrib.hooks.gcp_bigtable_hook.BigtableHook                                                              |airflow.gcp.hooks.bigtable.BigtableHook                                                                    |
 |airflow.contrib.hooks.gcp_cloud_build_hook.CloudBuildHook                                                         |airflow.gcp.hooks.cloud_build.CloudBuildHook                                                               |
-|airflow.contrib.hooks.gcp_compute_hook.GceHook                                                                    |airflow.gcp.hooks.compute.GceHook                                                                          |
+|airflow.contrib.hooks.gcp_compute_hook.GceHook                                                                    |airflow.gcp.hooks.compute.ComputeEngineOperator                                                                          |
```
Comment (TobKed) [inline UPDATING.md]: Shouldn't be `ComputeEngineHook ` instead of `ComputeEngineOperator`?
I am not sure is it allowed to change some older informations. Maybe it should be done in separate information. WDY @nuclearpinguin ?

File: airflow/contrib/hooks/gcp_compute_hook.py [hunk 2]
```diff
@@ -16,14 +16,30 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-"""This module is deprecated. Please use `airflow.gcp.hooks.compute`."""
+"""This module is deprecated. Please use `airflow.gcp.hooks.compute.ComputeEngineHook`."""
```
Comment (TobKed) [inline airflow/contrib/hooks/gcp_compute_hook.py]: I think information here shouldn't contain specific hook name, just path to module (as it was before).

File: airflow/contrib/hooks/gcp_compute_hook.py [hunk 3]
```diff
@@ -16,14 +16,30 @@
...
 
 import warnings
 
 # pylint: disable=unused-import
-from airflow.gcp.hooks.compute import GceHook, GceOperationStatus  # noqa
+from airflow.gcp.hooks.compute import ComputeEngineHook  # noqa
 
 warnings.warn(
-    "This module is deprecated. Please use `airflow.gcp.hooks.compute`.",
+    "This module is deprecated. airflow.gcp.hooks.compute.ComputeEngineHook`",
```
Comment (TobKed) [inline airflow/contrib/hooks/gcp_compute_hook.py]: Same as above.

**Review by TobKed — CHANGES_REQUESTED** [review]
Good job! I left few comments.

File: UPDATING.md (hunk 1 above)
Comment (mik-laj) [reply UPDATING.md]: It is recommended to change the old information if it is no longer correct. It is not advisable to make changes to UPDATING.MD in versions that have already been released, unless it corrects an error in the description. This note has been deliberately written in a manner that will allow the changes envisaged by AIP-21 to be made freely.

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 3 above)
Comment (mik-laj) [inline airflow/contrib/hooks/gcp_compute_hook.py]:
```suggestion
    "This module is deprecated. Please use `airflow.gcp.hooks.compute`.",
```

File: airflow/contrib/hooks/gcp_compute_hook.py [hunk 4]
```diff
@@ -16,14 +16,30 @@
...
     DeprecationWarning, stacklevel=2
 )
+
+
+class GceHook(ComputeEngineHook):
+    """
+    This module is deprecated.
```
Comment (mik-laj) [inline airflow/contrib/hooks/gcp_compute_hook.py]:
```suggestion
    This class is deprecated.
```

File: airflow/contrib/hooks/gcp_compute_hook.py [hunk 5]
```diff
@@ -16,14 +16,30 @@
...
+    Please use `airflow.gcp.hooks.compute.ComputeEngineHook`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        warnings.warn(
+            """This class is deprecated.
+            Please use `airflow.gcp.hooks.compute.ComputeEngineHook`.""",
+            DeprecationWarning,
```
Comment (mik-laj) [inline airflow/contrib/hooks/gcp_compute_hook.py]: Can you add stacklevel argument?

File: tests/gcp/hooks/test_gcp_compute_hook.py [hunk 6]
```diff
@@ -67,16 +67,15 @@ def test_start_instance_overridden_project_id(self, wait_for_operation_to_comple
         start_method.assert_called_once_with(instance='instance', project='example-project', zone='zone')
         execute_method.assert_called_once_with(num_retries=5)
         wait_for_operation_to_complete.assert_called_once_with(project_id='example-project',
-                                                               operation_name='operation_id',
```
Comment (mik-laj) [inline tests/gcp/hooks/test_gcp_compute_hook.py]: Can you remote this unrelated change?

File: tests/test_core_to_contrib.py [hunk 7]
```diff
@@ -16,787 +16,814 @@
...
 
 from parameterized import parameterized
 
+NEW_HOOK_PATHS = [
+    (
+        "airflow.gcp.hooks.compute.ComputeEngineHook",
+        "airflow.contrib.hooks.gcp_compute_hook.GceHook",
+    ),
+]
+OLD_HOOKS_PATHS = [
```
Comment (mik-laj) [inline tests/test_core_to_contrib.py]: What is the difference between this variable and the above?

File: UPDATING.md (hunk 1 above)
Comment (potiuk) [reply UPDATING.md]: I think it's quite justified to update "UPDATING.md" . It has not been released in 2.0 and we are not going to backport it to 1.10.* so it's ok to make it the "real" change here - even if we have to change it. We are going to change gcp to google soon and it's better to update UPDATING.md for that as well.

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 2 above)
Comment (potiuk) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Agree.

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 3 above)
Comment (potiuk) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Same!

Comment (potiuk) [reply airflow/contrib/hooks/gcp_compute_hook.py]: 👍

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 4 above)
Comment (potiuk) [reply airflow/contrib/hooks/gcp_compute_hook.py]: 👍

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 5 above)
Comment (potiuk) [reply airflow/contrib/hooks/gcp_compute_hook.py]: 👍

File: tests/test_core_to_contrib.py [hunk 8]
```diff
@@ -16,787 +16,814 @@
...
-        with self.assertWarns(DeprecationWarning):
+    def assert_proper_import(self, old_resource, new_resource):
+        new_path, _, _ = new_resource.rpartition(".")
+        old_path, _, _ = old_resource.rpartition(".")
+        with self.assertWarns(DeprecationWarning) as warning_msg:
             # Reload to see deprecation warning each time
-            old_module = importlib.reload(importlib.import_module(old_path))
-        new_module = importlib.import_module(new_path)
+            importlib.reload(importlib.import_module(old_path))
+            self.assert_warning(new_path, warning_msg)
```
Comment (potiuk) [inline tests/test_core_to_contrib.py:799]: I like that we assert proper warning message here !

**Review by potiuk — CHANGES_REQUESTED** [review]

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 4 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 3 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 2 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

File: tests/gcp/hooks/test_gcp_compute_hook.py (hunk 6 above)
Comment (michalslowikowski00) [reply tests/gcp/hooks/test_gcp_compute_hook.py]: Mistake, previous state was restored.

File: tests/test_core_to_contrib.py (hunk 7 above)
Comment (michalslowikowski00) [reply tests/test_core_to_contrib.py]: OLD_HOOKS_PATHS is variable which store paths that are not testable right now, NEW_HOOK_PATH is a place where I put paths that can be tested. With the end of the task OLD_HOOKS_PATHS will be removed permanently and NEW_HOOK_PATH will be replace with HOOK_PATH (just a var name for now), which will store all testable paths.

File: UPDATING.md (hunk 1 above)
Comment (michalslowikowski00) [reply UPDATING.md]: So, should I change the path in UPDATING.md, here https://github.com/apache/airflow/blob/master/UPDATING.md?

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 5 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

File: tests/test_core_to_contrib.py (hunk 8 above)
Comment (michalslowikowski00) [reply tests/test_core_to_contrib.py:799]: :)

File: tests/test_core_to_contrib.py (hunk 7 above)
Comment (mik-laj) [reply tests/test_core_to_contrib.py]: We discussed this changes offline.

**Comment by mik-laj** [comment]
Travis is sad. Can you fix it?

**Review by mik-laj — APPROVED** [review]
